### DISCUSSION TITLE
Exceptions a task can exhibit from missionary

### DISCUSSION THREAD

**Discussion by xificurC** [leonoel/missionary#57] [Q&A] [answered]
Your reClojure talk mentions catching an `InterruptedException` for a case when a task is interrupted. I see a [recent commit](https://github.com/leonoel/missionary/commit/07930608ff6dbf91dae1f51ee44df0d9612f1a61) that switches from `ExceptionInfo`s to a special `missionary.Cancelled` exception type. Can you elaborate what exceptions should a user of the library look for to handle and what would be the canonical way to handle them?

**Comment by leonoel** [comment, marked as answer]
`missionary` operators have the following general semantics :
* top-level effects throw `missionary.Cancelled` when cancelled before termination
* effect combinators propagate cancellation and errors untouched (including `via`, which delegates to the JVM thread interruption mechanism)

As mentioned in #51 interruption handling is a wild west, therefore `missionary` doesn't provide a generic catch-all-interruptions mechanism, the user must figure out how top-level effects react to cancellation and use `try`/`catch` accordingly. In the talk, the implementation of the task is not provided but it is assumed to be a third-party http library throwing `InterruptedException` on cancellation.
